### PR TITLE
Fix backend tests

### PR DESCRIPTION
**Pull request by floreks**
Rewritten tests disabled during dependency update.

### CODE REVIEW

**Comment by rf232** [comment]
Code looks fine, is it intentional though that the TestGetNodeDetail test from the nodedetail tests is not reenabled again?

**Comment by floreks** [comment]
>Code looks fine, is it intentional though that the TestGetNodeDetail test from the nodedetail tests is not reenabled again?

I was going through with the list from my previous PR. This test was not skipped and was not listed on my list and I've missed it :) Nice catch!

>I really love the simplifications with helper functions you made for creating stuff :)

I've had to go through core code to fix them and noticed similar approach. I've used that to simplify some tests a bit. It's hard to follow them sometimes :)

**Comment by floreks** [comment]
Done. PTAL.
